### PR TITLE
Remove unnecessary usages of Array

### PR DESCRIPTION
**Pull request by jreij**
## Description
Remove unnecessary usages of Array and replace them with List

## Checklist 
- [x] Code is unit tested
- [x] Changes are tested manually

COAND-717

### CODE REVIEW

File: card/src/main/java/com/adyen/checkout/card/util/CardNumberUtils.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            if (separatorIndexes.contains(index)) "$accumulation$separator$char" else accumulation + char
+        }
+
+        return splitStringWithMask(
+            string = unformattedString,
+            maskPartsLengths = maskPartsLengths
+        ).joinToString(separator)
+    }
+
+    private fun splitStringWithMask(string: String, maskPartsLengths: List<Int>): List<String> {
```
Comment (ozgur00) [inline card/src/main/java/com/adyen/checkout/card/util/CardNumberUtils.kt:27]: Did you leave this on purpose? I think the first part seems more readable to me. Only we could add some docs with examples in it.

File: card/src/main/java/com/adyen/checkout/card/util/CardNumberUtils.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+internal object CardNumberUtils {
+
+    fun formatCardNumber(unformattedString: String, maskPartsLengths: List<Int>, separator: String): String {
+        val separatorIndexes = List(maskPartsLengths.size) { index -> maskPartsLengths.take(index + 1).sum() }
+        return unformattedString.foldIndexed("") { index, accumulation, char ->
+            if (separatorIndexes.contains(index)) "$accumulation$separator$char" else accumulation + char
+        }
+
+        return splitStringWithMask(
```
Comment (OscarSpruit) [inline card/src/main/java/com/adyen/checkout/card/util/CardNumberUtils.kt:21]: Unreachable code detected

File: card/src/main/java/com/adyen/checkout/card/util/CardNumberUtils.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+package com.adyen.checkout.card.util
+
+import kotlin.math.min
+
+internal object CardNumberUtils {
+
+    fun formatCardNumber(unformattedString: String, maskPartsLengths: List<Int>, separator: String): String {
+        val separatorIndexes = List(maskPartsLengths.size) { index -> maskPartsLengths.take(index + 1).sum() }
+        return unformattedString.foldIndexed("") { index, accumulation, char ->
```
Comment (OscarSpruit) [inline card/src/main/java/com/adyen/checkout/card/util/CardNumberUtils.kt:17]: Oh wow this is beautiful!

File: card/src/main/java/com/adyen/checkout/card/ui/CardNumberInput.kt [hunk 4]
```diff
@@ -55,43 +55,31 @@ open class CardNumberInput @JvmOverloads constructor(
 
     override fun afterTextChanged(editable: Editable) {
         val initial = editable.toString()
-        var processed = initial.trim { it <= ' ' }.replace(DIGIT_SEPARATOR.toString().toRegex(), "")
+        var processed = initial.trim { it <= ' ' }.replace(DIGIT_SEPARATOR.toRegex(), "")
```
Comment (ozgur00) [inline card/src/main/java/com/adyen/checkout/card/ui/CardNumberInput.kt:58]: `.toRegex()` and `trim { }` is not needed here

**Review by ozgur00 — APPROVED** [review]

**Pull request merged**